### ISSUE TITLE
Ignore repos list

### ISSUE THREAD

**Issue by sushain97** [apertium/phenny#395]
There should be a config option for which repos to ignore events from. `apertium-all`, `apertium-trunk`, etc.

**Comment by ftyers** [comment]
This seems fairly straightforward in git.py line 150. `do_POST_unsafe(self, data):`

**Comment by Androbin** [comment]
Current behavior:
* `if master and not fork` -> default channel
* `else` -> branch channels

**Comment by Androbin** [comment]
If we want to ignore all events from repo X, why don't we just disable the event hook in the repo settings?

**Comment by sushain97** [comment]
> If we want to ignore all events from repo X, why don't we just disable the event hook in the repo settings?

That isn't possible, unfortunately. The webhook is at the org level and it will continue to be.

**Comment by Androbin** [comment]
Okay, get it. What parameters do we want? When shall messages be sent to which channels?

**Comment by Androbin** [comment]
Shall we leave the current logic in place and just filter for repositories in e.g. `git_repositories`?

**Comment by Androbin** [comment]
I guess we want to apply this to the entire hook, not just PUSH events?

**Comment by ftyers** [comment]
The ideal thing would be to have a per-channel, per-repo whitelist/blacklist of events.

E.g. on `#apertium` only publish:

whitelist:
* new issues
* push 
blacklist
* everything else
* push apertium-all, apertium-incubator, apertium-trunk, ...

**Comment by Androbin** [comment]
@ftyers We need to bring that into this form:
( repo_name, branch_name, is_fork, event_type ) -> list_of_channels

**Comment by Androbin** [comment]
@jonorthwash See #396 for that.

**Comment by sushain97** [comment]
No, ours should be more like:

```
git_events = {
  '*': ['push'],
  'apertium/*': ['issue', 'push'],
  'apertium/apertium-all': [],
  'apertium/apertium-incubator': [],
  'apertium/apertium-nursery': [],
  'apertium/apertium-staging': [],
  'apertium/apertium-trunk': [],
  'apertium/apertium-tools': [],
  'apertium/apertium-incubator': [],
}
```

**Comment by jonorthwash** [comment]
We're still getting these commits:
```
(14:56:21) begiak: apertium-incubator: sushain97 * apertium-byv-fra: Sync submodules (1U, 0D, 0A) Updated: apertium-byv-fra. Deleted: None. Added: None. https://github.com/apertium/apertium-incubator/commit/2d4701
```
and
```
(14:56:36) begiak: apertium-all: sushain97 * apertium-incubator: Sync submodules (1U, 0D, 0A) Updated: apertium-incubator. Deleted: None. Added: None. https://github.com/apertium/apertium-all/commit/4bf649
```

**Comment by jonorthwash** [comment]
This is what I have in the config:
```
git_events = {
    '*': ['push'],
    'apertium/*': ['issue', 'push'],
    'apertium/apertium-all': [],
    'apertium/apertium-incubator': [],
    'apertium/apertium-nursery': [],
    'apertium/apertium-staging': [],
    'apertium/apertium-trunk': [],
    'apertium/apertium-tools': [],
}
```

**Comment by Androbin** [comment]
Oops, I know what's wrong there...

**Comment by Androbin** [comment]
Fixed in 36453f54494cfd3a8b57f974d05845a08d55d3e7
